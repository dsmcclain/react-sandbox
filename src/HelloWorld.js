import React, { Component } from 'react';
import './HelloWorld.css';

class HelloWorld extends Component {
  constructor(props) {
      super(props);
      this.state = { 
        greeting: 'Hello',
        fredCount: 0,
      };
      this.fredify = this.fredify.bind(this);
      this.removePeople = this.removePeople.bind(this);
  }

  fredify() {
      const fredness = this.state.greeting;
      if (fredness === 'Hello') {
        this.setState({ greeting: 'Yaba Daba Doo' });
        this.setState({ fredCount: this.state.fredCount + 1 });
      } else {
        this.setState({ greeting: 'Hello' });
      };
  }

  removePeople() {
      this.props.removePeople(this.props.name);
  }
  
  render() {
    return (
      <div className="HelloWorld">
        {this.state.greeting} {this.props.name}!
        <br />
        You have been Fredified {this.state.fredCount} times.
        <br />
        <button className="fredify" onClick={this.fredify}>Fredify!</button>
        <button className="remove" onClick={this.removePeople}>Remove Me!</button>
      </div>
    );
  }
}

export default HelloWorld;