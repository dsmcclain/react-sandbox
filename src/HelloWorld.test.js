import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import HelloWorld from './HelloWorld';

describe(HelloWorld, () => {
    const name = 'Person';
    const mockRemovePeople = jest.fn();
    const component = shallow(
        <HelloWorld name={name} removePeople={mockRemovePeople}/>
    );

    it ('renders and matches our snapshot', () => {
        const component = renderer.create(
            <HelloWorld name="Person" />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('contains the supplied name', () => {
        expect(component.text()).toContain(name);
    });

    it('modifies the greeting when fredify button is clicked', () => {
        component.find('button.fredify').simulate('click');
        expect(component.text()).toContain('Yaba Daba Doo');
    });

    it('calls the passed in removePeople function when remove button is clicked', () => {
        component.find('button.remove').simulate('click');
        expect(mockRemovePeople).toBeCalled();
    });
});