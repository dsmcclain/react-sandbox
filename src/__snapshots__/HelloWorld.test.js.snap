// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld renders and matches our snapshot 1`] = `
<div
  className="HelloWorld"
>
  Hello
   
  Person
  !
  <br />
  <button
    className="fredify"
    onClick={[Function]}
  >
    Fredify!
  </button>
  <button
    className="remove"
    onClick={[Function]}
  >
    Remove Me!
  </button>
</div>
`;
